src/WebJobs.Script/Workers/ProcessManagement/WorkerProcess.cs

```diff
@@ -84,52 +86,46 @@ public ProcessStats GetStats()
 
         private void OnErrorDataReceived(object sender, DataReceivedEventArgs e)
         {
-            // TODO: per language stdout/err parser?
             if (e.Data != null)
             {
-                string msg = e.Data;
-                if (msg.IndexOf("warn", StringComparison.OrdinalIgnoreCase) > -1)
+                ParseConsoleLog(e.Data, true);
+            }
+        }
+
+        internal void ParseConsoleLog(string msg, bool errorData = false)
+        {
+            if (msg.IndexOf("warn", StringComparison.OrdinalIgnoreCase) > -1)
+            {
+                BuildAndLogConsoleLog(msg, LogLevel.Warning);
+            }
+            else
+            {
+                if (_useStdErrorStreamForErrorsOnly && errorData)
                 {
-                    if (WorkerProcessUtilities.IsConsoleLog(msg))
-                    {
-                        msg = WorkerProcessUtilities.RemoveLogPrefix(msg);
-                        _workerProcessLogger?.LogWarning(msg);
-                    }
-                    else
-                    {
-                        _consoleLogSource?.Log(msg);
-                    }
-                }
-                else if ((msg.IndexOf("error", StringComparison.OrdinalIgnoreCase) > -1) ||
-                          (msg.IndexOf("fail", StringComparison.OrdinalIgnoreCase) > -1) ||
-                          (msg.IndexOf("severe", StringComparison.OrdinalIgnoreCase) > -1))
-                {
-                    if (WorkerProcessUtilities.IsConsoleLog(msg))
-                    {
-                        msg = WorkerProcessUtilities.RemoveLogPrefix(msg);
-                        _workerProcessLogger?.LogError(msg);
-                    }
-                    else
-                    {
-                        _consoleLogSource?.Log(msg);
-                    }
-                    _processStdErrDataQueue = WorkerProcessUtilities.AddStdErrMessage(_processStdErrDataQueue, Sanitizer.Sanitize(msg));
+                    LogError(msg);
                 }
                 else
                 {
-                    if (WorkerProcessUtilities.IsConsoleLog(msg))
+                    if ((msg.IndexOf("error", StringComparison.OrdinalIgnoreCase) > -1) ||
+                              (msg.IndexOf("fail", StringComparison.OrdinalIgnoreCase) > -1) ||
+                              (msg.IndexOf("severe", StringComparison.OrdinalIgnoreCase) > -1))
```

Comment:
Same question about `errorData == true` here. Ex: if a user does `console.log("No errors found");` in the node worker, it would be marked as an error?

Reply:
you are right. Thanks for careful review. Updated code to skip parsing logic for stdout which is the case today as well.